docs/reference/api.rst

```diff
@@ -384,36 +405,78 @@ end and last_id
 
     The return value of these methods that support paging contains a ``links`` tag, with the urls of the ``next`` and
     ``prev`` pages. To iterate over all the results, the client can follow these links, or alternatively call the
-    ``all()`` method on the result:
+    helper methods on the :py:class:`inmanta.protocol.common.PageableResult` object:
 
     .. code-block:: python
 
         # Iterate over all results by fetching pages of size 20
 
-        result = await client.resource_list(tid=env.id, limit=20)
+        async for item in client.resource_list(tid=env.id, limit=20).all():
+            ...
 
-        async for item in result.all():
+
+        # Iterate over results on a single page
+
+        for item in await client.resource_list(tid=env.id, limit=20).value():
             ...
 
 
+    When calling an endpoint from a synchronous context, the :py:meth:`inmanta.protocol.common.PageableResult.all_sync`
+    method should be used instead of the all() method:
+
+    .. code-block:: python
+
+        # Iterate over all results by fetching pages of size 20
+        for item in client.resource_list(tid=env.id, limit=20).all_sync():
+            assert item == all_resources[idx]
+            idx += 1
+
+
+.. _python_client_mypy_plugin:
+
+.. note::
+
+    Is it possible to inspect typing information about the :py:class:`inmanta.protocol.common.Client` and the
```

Comment:
I would find it more natural to mention that we can do type checks on the result with the plugin first and how to set it up?